<?xml version="1.0" ?>

<launch>
    <!-- visualization config -->
    <arg name="rviz"               default="true" />
    <arg name="description_name"   default="legged_robot_description"/>
    <arg name="multiplot"          default="false"/>

    <!-- The task file for the mpc. -->
    <arg name="taskFile"          default="$(find ocs2_legged_robot)/config/mpc/task.info"/>
    <!-- The reference related config file of the robot -->
    <arg name="referenceFile"     default="$(find ocs2_legged_robot)/config/command/reference.info"/>
    <!-- The URDF model of the robot -->
    <arg name="urdfFile"          default="$(find ocs2_robotic_assets)/resources/anymal_c/urdf/anymal.urdf"/>
    <!-- The file defining gait definition -->
    <arg name="gaitCommandFile"   default="$(find ocs2_legged_robot)/config/command/gait.info"/>

    <!-- rviz -->
    <group if="$(arg rviz)">
      <param name="$(arg description_name)" textfile="$(arg urdfFile)"/>
      <arg name="rvizconfig" default="$(find ocs2_legged_robot_ros)/rviz/legged_robot.rviz" />
      <node pkg="rviz" type="rviz" name="rviz" args="-d $(arg rvizconfig)" output="screen" />
    </group>

    <!-- multiplot -->
    <group if="$(arg multiplot)">
      <include file="$(find ocs2_legged_robot_ros)/launch/multiplot.launch"/>
    </group>

    <!-- make the files into global parameters -->
    <param name="multiplot"         value="$(arg multiplot)"/>
    <param name="taskFile"          value="$(arg taskFile)" />
    <param name="referenceFile"     value="$(arg referenceFile)" />
    <param name="urdfFile"          value="$(arg urdfFile)" />
    <param name="gaitCommandFile"   value="$(arg gaitCommandFile)"/>

    <node pkg="ocs2_legged_robot_ros" type="legged_robot_sqp_mpc" name="legged_robot_sqp_mpc"
          output="screen" launch-prefix=""/>

    <node pkg="ocs2_legged_robot_ros" type="legged_robot_dummy" name="legged_robot_dummy"
          output="screen" launch-prefix="gnome-terminal --"/>

    <node pkg="ocs2_legged_robot_ros" type="legged_robot_target" name="legged_robot_target"
          output="screen" launch-prefix="gnome-terminal --"/>

    <node pkg="ocs2_legged_robot_ros" type="legged_robot_gait_command" name="legged_robot_gait_command"
          output="screen" launch-prefix="gnome-terminal --"/>
</launch>
